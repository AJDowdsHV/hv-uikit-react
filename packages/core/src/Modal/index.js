export { default } from "./Modal";
export { default as HvModalTitle } from "./ModalTitle";
export { default as HvModalContent } from "./ModalContent";
export { default as HvModalActions } from "./ModalActions";
