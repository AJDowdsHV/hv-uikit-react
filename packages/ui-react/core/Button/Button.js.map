{"version":3,"sources":["../../src/Button/Button.js"],"names":["HvButton","props","classes","children","disabled","onClick","root","propTypes","PropTypes","instanceOf","Object","isRequired","string","bool","Function","defaultProps"],"mappings":";;;;;;;;;AAUA;;AACA;;AACA;;AAZA;;;;;;;;;AAcA,IAAMA,QAAQ,GAAG,SAAXA,QAAW,CAAAC,KAAK,EAAI;AAAA,MAChBC,OADgB,GACyBD,KADzB,CAChBC,OADgB;AAAA,MACPC,QADO,GACyBF,KADzB,CACPE,QADO;AAAA,MACGC,QADH,GACyBH,KADzB,CACGG,QADH;AAAA,MACaC,OADb,GACyBJ,KADzB,CACaI,OADb;AAGxB,SACE,6BAAC,eAAD;AAAQ,IAAA,SAAS,EAAEH,OAAO,CAACI,IAA3B;AAAiC,IAAA,QAAQ,EAAEF,QAA3C;AAAqD,IAAA,aAAa,MAAlE;AAAmE,IAAA,OAAO,EAAEC;AAA5E,KACGF,QADH,CADF;AAKD,CARD;;AAUAH,QAAQ,CAACO,SAAT,GAAqB;AACnBL,EAAAA,OAAO,EAAEM,mBAAUC,UAAV,CAAqBC,MAArB,EAA6BC,UADnB;AAEnBR,EAAAA,QAAQ,EAAEK,mBAAUI,MAAV,CAAiBD,UAFR;AAGnBP,EAAAA,QAAQ,EAAEI,mBAAUK,IAHD;AAInBR,EAAAA,OAAO,EAAEG,mBAAUC,UAAV,CAAqBK,QAArB;AAJU,CAArB;AAOAd,QAAQ,CAACe,YAAT,GAAwB;AACtBX,EAAAA,QAAQ,EAAE;AADY,CAAxB;eAIeJ,Q","sourcesContent":["/**\n * Copyright (c) 2018 Hitachi Vantara Corporation.\n *\n *  The copyright to the computer software herein is the property of\n *  Hitachi Vantara Corporation. The software may be used and/or copied only\n *  with the written permission of Hitachi Vantara Corporation or in accordance\n *  with the terms and conditions stipulated in the agreement/contract\n *  under which the software has been supplied.\n */\n\nimport React from \"react\";\nimport PropTypes from \"prop-types\";\nimport Button from \"@material-ui/core/Button\";\n\nconst HvButton = props => {\n  const { classes, children, disabled, onClick } = props;\n\n  return (\n    <Button className={classes.root} disabled={disabled} disableRipple onClick={onClick}>\n      {children}\n    </Button>\n  );\n};\n\nHvButton.propTypes = {\n  classes: PropTypes.instanceOf(Object).isRequired,\n  children: PropTypes.string.isRequired,\n  disabled: PropTypes.bool,\n  onClick: PropTypes.instanceOf(Function)\n};\n\nHvButton.defaultProps = {\n  disabled: false\n};\n\nexport default HvButton;\n"],"file":"Button.js"}