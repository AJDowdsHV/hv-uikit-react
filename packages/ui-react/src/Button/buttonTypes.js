const buttonTypes = Object.freeze({
    primary: "primary",
    secondary: "secondary",
    link: "link",
    inspireLink: "inspireLink"
});

export default buttonTypes;