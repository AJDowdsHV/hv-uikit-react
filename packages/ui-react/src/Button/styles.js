/**
 * Copyright (c) 2018 Hitachi Vantara Corporation.
 *
 *  The copyright to the computer software herein is the property of
 *  Hitachi Vantara Corporation. The software may be used and/or copied only
 *  with the written permission of Hitachi Vantara Corporation or in accordance
 *  with the terms and conditions stipulated in the agreement/contract
 *  under which the software has been supplied.
 */

const styles = theme => ({
  root: {
    textTransform: "capitalize",
    "&:hover,&:focus": {},
    "&:active": {},
    minWidth: "70px",
    padding: "0 10px",
    cursor: "pointer",
    height: "32px",
    fontWeight: theme.hv.typography.buttonText.fontWeight,
    letterSpacing: theme.hv.typography.buttonText.letterSpacing
  },
  containedPrimary: {
    color: theme.palette.common.white,
    backgroundColor: theme.palette.primary.main,
    "&:hover": {
      backgroundColor: theme.palette.primary.light
    },
    "&:active": {
      backgroundColor: theme.palette.primary.main
    },
    "&$disabled": {
      backgroundColor: theme.palette.grey.clear,
      color: theme.palette.grey.passive,
      cursor: "not-allowed",
      pointerEvents: "auto"
    },
    "&$disabled&:hover": {
      backgroundColor: theme.palette.grey.clear,
      color: theme.palette.grey.passive,
      cursor: "not-allowed",
      pointerEvents: "auto"
    }
  },
  outlinedPrimary: {
    backgroundColor: theme.palette.common.white,
    color: theme.palette.grey.inspire,
    border: `1px solid ${theme.palette.grey.plain}`,
    "&:hover": {
      backgroundColor: theme.palette.grey.rainy,
      border: `1px solid ${theme.palette.grey.plain}`,
    },
    "&:active": {
      backgroundColor: theme.palette.common.white
    },
    "&$disabled": {
      backgroundColor: theme.palette.grey.clear,
      color: theme.palette.grey.passive,
      border: "none",
      cursor: "not-allowed",
      pointerEvents: "auto"
    },
    "&$disabled&:hover": {
      backgroundColor: theme.palette.grey.clear,
      color: theme.palette.grey.passive,
      border: "none",
      cursor: "not-allowed",
      pointerEvents: "auto"
    } 
  },
  textPrimary: {
    color: theme.palette.primary.main,
    backgroundColor: theme.palette.common.white,
    "&:hover": {
      backgroundColor: theme.palette.grey.foggy,
    },
    "&:active": {
      backgroundColor: theme.palette.common.white
    },
    "&$disabled": {
      backgroundColor: theme.palette.common.white,
      color: theme.palette.grey.passive,
      border: "none",
      cursor: "not-allowed",
      pointerEvents: "auto"
    },
    "&$disabled&:hover": {
      backgroundColor: theme.palette.common.white,
      color: theme.palette.grey.passive,
      border: "none",
      cursor: "not-allowed",
      pointerEvents: "auto"
    }
  },
  disabled: {
    backgroundColor: theme.palette.grey.clear,
    color: theme.palette.grey.passive,
    cursor: "not-allowed",
  }
});

export default styles;
