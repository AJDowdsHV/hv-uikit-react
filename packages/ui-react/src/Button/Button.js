/**
 * Copyright (c) 2018 Hitachi Vantara Corporation.
 *
 *  The copyright to the computer software herein is the property of
 *  Hitachi Vantara Corporation. The software may be used and/or copied only
 *  with the written permission of Hitachi Vantara Corporation or in accordance
 *  with the terms and conditions stipulated in the agreement/contract
 *  under which the software has been supplied.
 */

import React from "react";
import PropTypes from "prop-types";
import Button from "@material-ui/core/Button";
import buttonTypes from "./buttonTypes";
import materialButtonConfiguration from "./materialButtonConfiguration";

/**
 * Receives a type of HvButton and generates an appropiate material ui configuration.
 *
 * @param {*} type - The button type configuration that will be generated.
 * @returns {Object} - An Object with the color and variant values required by the material ui button.
 */
const generateButtonConfiguration = type => {
  switch (type) {
    default:
    case buttonTypes.primary:
      return {
        color: materialButtonConfiguration.color.primary,
        variant: materialButtonConfiguration.variant.contained
      };
    case buttonTypes.secondary:
      return {
        color: materialButtonConfiguration.color.primary,
        variant: materialButtonConfiguration.variant.outlined
      };
    case buttonTypes.link:
      return {
        color: materialButtonConfiguration.color.primary,
        variant: materialButtonConfiguration.variant.text
      };
  }
};

const HvButton = props => {
  const {
    classes,
    className,
    children,
    disabled,
    onClick,
    type,
    colorType,
    ...other
  } = props;
  const buttonConfiguration = generateButtonConfiguration(colorType);

  return (
    <Button
      className={className}
      classes={classes}
      variant={buttonConfiguration.variant}
      color={buttonConfiguration.color}
      disabled={disabled}
      disableRipple
      onClick={onClick}
      type={type}
      {...other}
    >
      {children}
    </Button>
  );
};

HvButton.propTypes = {
  /**
   * Type of button to use.
   *  - Accepted values:
   *    --"submit",
   *    --"reset",
   *    --"button"
   */
  type: PropTypes.oneOf(["submit", "reset", "button"]),
  /**
   * Type of color of HvButton to use.
   *  - Accepted values:
   *    --"primary",
   *    --"secondary",
   *    --"link"
   *  - note: the buttonType object should be used to set this value.
   */
  colorType: PropTypes.oneOf(["primary", "secondary", "link"]),
  /**
   * Class names to be applied.
   */
  className: PropTypes.string,
  /**
   * A Jss Object used to override or extend the styles applied to the button.
   */
  classes: PropTypes.instanceOf(Object).isRequired,
  /**
   * The content inside the button.
   */
  children: PropTypes.string.isRequired,
  /**
   * If set to `true` the button is disabled and the onClick function will not be called.
   */
  disabled: PropTypes.bool,
  /**
   * The function executed when the button is pressed.
   */
  onClick: PropTypes.instanceOf(Function)
};

HvButton.defaultProps = {
  className: "",
  type: "button",
  colorType: buttonTypes.Primary,
  disabled: false,
  onClick: () => {}
};

export default HvButton;
