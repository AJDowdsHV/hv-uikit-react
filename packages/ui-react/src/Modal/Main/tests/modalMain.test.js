/**
 * Copyright (c) 2018 Hitachi Vantara Corporation.
 *
 *  The copyright to the computer software herein is the property of
 *  Hitachi Vantara Corporation. The software may be used and/or copied only
 *  with the written permission of Hitachi Vantara Corporation or in accordance
 *  with the terms and conditions stipulated in the agreement/contract
 *  under which the software has been supplied.
 */

/* eslint-env jest */

import React from "react";
import { shallow, mount } from "enzyme";

import Backdrop from "@material-ui/core/Backdrop";

import HvProvider from "../../../Provider";
import Button from "../../../Button";

import ModalWithStyles from "../index";
import Modal from "../Main";

describe("Modal withStyles", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = shallow(
      <HvProvider>
        <ModalWithStyles>Modal Content</ModalWithStyles>
      </HvProvider>
    );
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Modal Component", () => {
  let wrapper;
  const onCloseMock = jest.fn();
  const open = true;

  beforeEach(async () => {
    wrapper = shallow(
      <Modal classes={{}} open={open} onClose={onCloseMock}>
        Modal Content
      </Modal>
    );
    onCloseMock.mockClear();
  });

  it("should render correctly if opened", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly if closed", () => {
    wrapper = shallow(
      <Modal classes={{}} open={false} onClose={onCloseMock}>
        Modal Content
      </Modal>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("allows external props to be added", () => {
    wrapper = shallow(
      <Modal
        classes={{}}
        open={open}
        onClose={onCloseMock}
        disableBackdropClick
        disableEscapeKeyDown
      >
        Modal Content
      </Modal>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("allows external styles to be added", () => {
    wrapper = shallow(
      <Modal
        classes={{
          root: "testClassRoot",
          paper: "testClassPaper",
          background: "testClassBackground",
          closeButton: "testClassCloseButton"
        }}
        open={open}
        onClose={onCloseMock}
      >
        Modal Content
      </Modal>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("onClose should be called when close button is clicked", () => {
    const button = wrapper.find(Button);
    button.at(0).simulate("click");
    expect(onCloseMock).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });

  it("onClose should be called when close is triggered in the dialog", () => {
    wrapper = mount(
      <HvProvider>
        <Modal classes={{}} open={open} onClose={onCloseMock}>
          Modal Content
        </Modal>
      </HvProvider>
    );

    const backdrop = wrapper.find(Backdrop);
    backdrop.at(0).simulate("click");
    expect(onCloseMock).toHaveBeenCalled();
  });
});
