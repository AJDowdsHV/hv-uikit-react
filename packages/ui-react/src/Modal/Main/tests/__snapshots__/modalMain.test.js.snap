// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component allows external props to be added 1`] = `
<WithStyles(Dialog)
  BackdropProps={
    Object {
      "classes": Object {
        "root": undefined,
      },
    }
  }
  PaperProps={
    Object {
      "classes": Object {
        "root": undefined,
      },
    }
  }
  className=""
  disableBackdropClick={true}
  disableEscapeKeyDown={true}
  onClose={[Function]}
  open={true}
>
  <WithStyles(HvButton)
    colorType="link"
    onClick={[Function]}
  >
    <Close16
      style={
        Object {
          "width": "32px",
        }
      }
    />
  </WithStyles(HvButton)>
  Modal Content
</WithStyles(Dialog)>
`;

exports[`Modal Component allows external styles to be added 1`] = `
<WithStyles(Dialog)
  BackdropProps={
    Object {
      "classes": Object {
        "root": "testClassBackground",
      },
    }
  }
  PaperProps={
    Object {
      "classes": Object {
        "root": "testClassPaper",
      },
    }
  }
  className="testClassRoot"
  onClose={[Function]}
  open={true}
>
  <WithStyles(HvButton)
    className="testClassCloseButton"
    colorType="link"
    onClick={[Function]}
  >
    <Close16
      style={
        Object {
          "width": "32px",
        }
      }
    />
  </WithStyles(HvButton)>
  Modal Content
</WithStyles(Dialog)>
`;

exports[`Modal Component onClose should be called when close button is clicked 1`] = `
<WithStyles(Dialog)
  BackdropProps={
    Object {
      "classes": Object {
        "root": undefined,
      },
    }
  }
  PaperProps={
    Object {
      "classes": Object {
        "root": undefined,
      },
    }
  }
  className=""
  onClose={[Function]}
  open={true}
>
  <WithStyles(HvButton)
    colorType="link"
    onClick={[Function]}
  >
    <Close16
      style={
        Object {
          "width": "32px",
        }
      }
    />
  </WithStyles(HvButton)>
  Modal Content
</WithStyles(Dialog)>
`;

exports[`Modal Component should render correctly if closed 1`] = `
<WithStyles(Dialog)
  BackdropProps={
    Object {
      "classes": Object {
        "root": undefined,
      },
    }
  }
  PaperProps={
    Object {
      "classes": Object {
        "root": undefined,
      },
    }
  }
  className=""
  onClose={[Function]}
  open={false}
>
  <WithStyles(HvButton)
    colorType="link"
    onClick={[Function]}
  >
    <Close16
      style={
        Object {
          "width": "32px",
        }
      }
    />
  </WithStyles(HvButton)>
  Modal Content
</WithStyles(Dialog)>
`;

exports[`Modal Component should render correctly if opened 1`] = `
<WithStyles(Dialog)
  BackdropProps={
    Object {
      "classes": Object {
        "root": undefined,
      },
    }
  }
  PaperProps={
    Object {
      "classes": Object {
        "root": undefined,
      },
    }
  }
  className=""
  onClose={[Function]}
  open={true}
>
  <WithStyles(HvButton)
    colorType="link"
    onClick={[Function]}
  >
    <Close16
      style={
        Object {
          "width": "32px",
        }
      }
    />
  </WithStyles(HvButton)>
  Modal Content
</WithStyles(Dialog)>
`;

exports[`Modal withStyles should render correctly 1`] = `
<MuiThemeProviderOld
  sheetsManager={Map {}}
  theme={
    Object {
      "breakpoints": Object {
        "between": [Function],
        "down": [Function],
        "keys": Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ],
        "only": [Function],
        "up": [Function],
        "values": Object {
          "lg": 1280,
          "md": 960,
          "sm": 600,
          "xl": 1920,
          "xs": 0,
        },
        "width": [Function],
      },
      "direction": "ltr",
      "hv": Object {
        "palette": Object {
          "common": Object {
            "black": "#000000",
            "white": "#FFFFFF",
          },
          "grey": Object {
            "clear": "#F0F0F0",
            "foggy": "#F5F5F5",
            "inspire": "#414141",
            "passive": "#999999",
            "plain": "#BCBCBC",
            "rainy": "#DEDEDE",
            "smokey": "#F9F9F9",
          },
          "hitachi": Object {
            "light": "#FF312E",
            "main": "#CC0000",
          },
          "primary": Object {
            "light": "#4389DB",
            "main": "#146BD2",
          },
          "secondary": Object {
            "light": "#999999",
            "main": "#414141",
          },
          "severity": Object {
            "critical": "#F9E7CF",
            "error": "#F5D8D8",
            "minor": "#E2F4F4",
            "negligible": "#F8F8F8",
            "success": "#DFECD1",
            "warning": "#F7F2D6",
          },
          "status": Object {
            "critical": "#E68C17",
            "error": "#D13F3F",
            "minor": "#72CCCB",
            "negligible": "#DEDEDE",
            "success": "#63A621",
            "warning": "#D9C138",
          },
          "support": Object {
            "darkTurquoise": "#0F8B8D",
            "lightTurquoise": "#C3ECEC",
            "orange": "#FFBB00",
            "plum": "#7C0B2B",
            "redOrange": "#FF5838",
            "turquoise": "#7CDEDC",
            "yellow": "#F9DC33",
          },
          "text": Object {
            "primary": "#414141",
            "secondary": "#999999",
          },
        },
        "spacing": Object {
          "lg": "60",
          "md": "30",
          "sm": "20",
          "xl": "90",
          "xs": "10",
        },
        "typography": Object {
          "buttonText": Object {
            "color": "#FFFFFF",
            "fontSize": "14px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "21px",
          },
          "disabledButtonText": Object {
            "color": "#999999",
            "fontSize": "12px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "16px",
          },
          "disabledText": Object {
            "color": "#999999",
            "fontSize": "12px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "16px",
          },
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "highlightText": Object {
            "color": "#414141",
            "fontSize": "14px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "20px",
          },
          "infoText": Object {
            "color": "#414141",
            "fontSize": "12px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "16px",
          },
          "inlineLink": Object {
            "color": "#146BD2",
            "fontSize": "14px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "20px",
          },
          "labelText": Object {
            "color": "#414141",
            "fontSize": "12px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "16px",
          },
          "largeTitle": Object {
            "color": "#414141",
            "fontSize": "32px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "40px",
          },
          "mediumTitle": Object {
            "color": "#414141",
            "fontSize": "22px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "30px",
          },
          "normalText": Object {
            "color": "#414141",
            "fontSize": "14px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "20px",
          },
          "sliderTrackText": Object {
            "color": "#414141",
            "fontSize": "10px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "12px",
          },
          "smallTitle": Object {
            "color": "#414141",
            "fontSize": "22px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "30px",
          },
          "visualizationText": Object {
            "color": "#414141",
            "fontSize": "10px",
            "fontWeight": "400",
            "letterSpacing": "0.02em",
            "lineHeight": "15px",
          },
          "xlargeTitle": Object {
            "color": "#414141",
            "fontSize": "32px",
            "fontWeight": "600",
            "letterSpacing": "0.02em",
            "lineHeight": "40px",
          },
        },
      },
      "mixins": Object {
        "gutters": [Function],
        "toolbar": Object {
          "@media (min-width:0px) and (orientation: landscape)": Object {
            "minHeight": 48,
          },
          "@media (min-width:600px)": Object {
            "minHeight": 64,
          },
          "minHeight": 56,
        },
      },
      "overrides": Object {
        "MuiAppBar": Object {
          "colorDefault": Object {
            "backgroundColor": "#FFFFFF",
            "contrastText": "#414141",
          },
          "root": Object {
            "@media (min-width: 600px)": Object {
              "height": "50px",
            },
            "borderTop": "4px solid #CC0000",
          },
        },
        "MuiToolbar": Object {
          "dense": Object {
            "minHeight": "46px",
          },
          "gutters": Object {
            "@media (min-width:600px)": Object {
              "paddingLeft": "20px",
              "paddingRight": "20px",
            },
            "paddingLeft": "20px",
            "paddingRight": "20px",
          },
          "root": Object {
            "@media (min-width: 600px)": Object {
              "minHeight": "46px",
            },
            "alignItems": "center",
          },
        },
      },
      "palette": Object {
        "action": Object {
          "active": "rgba(0, 0, 0, 0.54)",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "disabledBackground": "rgba(0, 0, 0, 0.12)",
          "hover": "rgba(0, 0, 0, 0.08)",
          "hoverOpacity": 0.08,
          "selected": "rgba(0, 0, 0, 0.14)",
        },
        "augmentColor": [Function],
        "background": Object {
          "default": "#F9F9F9",
          "paper": "#fff",
        },
        "common": Object {
          "black": "#000000",
          "white": "#FFFFFF",
        },
        "contrastThreshold": 3,
        "divider": "rgba(0, 0, 0, 0.12)",
        "error": Object {
          "contrastText": "#fff",
          "dark": "#d32f2f",
          "light": "#e57373",
          "main": "#f44336",
        },
        "getContrastText": [Function],
        "grey": Object {
          "100": "#f5f5f5",
          "200": "#eeeeee",
          "300": "#e0e0e0",
          "400": "#bdbdbd",
          "50": "#fafafa",
          "500": "#9e9e9e",
          "600": "#757575",
          "700": "#616161",
          "800": "#424242",
          "900": "#212121",
          "A100": "#d5d5d5",
          "A200": "#aaaaaa",
          "A400": "#303030",
          "A700": "#616161",
          "clear": "#F0F0F0",
          "foggy": "#F5F5F5",
          "inspire": "#414141",
          "passive": "#999999",
          "plain": "#BCBCBC",
          "rainy": "#DEDEDE",
          "smokey": "#F9F9F9",
        },
        "hitachi": Object {
          "light": "#FF312E",
          "main": "#CC0000",
        },
        "primary": Object {
          "contrastText": "#fff",
          "dark": "rgb(14, 74, 147)",
          "light": "#4389DB",
          "main": "#146BD2",
        },
        "secondary": Object {
          "contrastText": "#fff",
          "dark": "rgb(45, 45, 45)",
          "light": "#999999",
          "main": "#414141",
        },
        "severity": Object {
          "critical": "#F9E7CF",
          "error": "#F5D8D8",
          "minor": "#E2F4F4",
          "negligible": "#F8F8F8",
          "success": "#DFECD1",
          "warning": "#F7F2D6",
        },
        "status": Object {
          "critical": "#E68C17",
          "error": "#D13F3F",
          "minor": "#72CCCB",
          "negligible": "#DEDEDE",
          "success": "#63A621",
          "warning": "#D9C138",
        },
        "support": Object {
          "darkTurquoise": "#0F8B8D",
          "lightTurquoise": "#C3ECEC",
          "orange": "#FFBB00",
          "plum": "#7C0B2B",
          "redOrange": "#FF5838",
          "turquoise": "#7CDEDC",
          "yellow": "#F9DC33",
        },
        "text": Object {
          "disabled": "#999999",
          "hint": "rgba(0, 0, 0, 0.38)",
          "primary": "#414141",
          "secondary": "rgba(0, 0, 0, 0.54)",
        },
        "tonalOffset": 0.2,
        "type": "light",
      },
      "props": Object {
        "MuiButtonBase": Object {
          "disableRipple": true,
        },
        "MuiInput": Object {
          "disableUnderline": true,
        },
      },
      "shadows": Array [
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
        "none",
      ],
      "shape": Object {
        "borderRadius": 0,
      },
      "spacing": Object {
        "lg": "60",
        "md": "30",
        "sm": "20",
        "unit": 8,
        "xl": "90",
        "xs": "10",
      },
      "transitions": Object {
        "create": [Function],
        "duration": Object {
          "complex": 375,
          "enteringScreen": 225,
          "leavingScreen": 195,
          "short": 250,
          "shorter": 200,
          "shortest": 150,
          "standard": 300,
        },
        "easing": Object {
          "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
          "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
          "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
          "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
        },
        "getAutoHeightDuration": [Function],
      },
      "typography": Object {
        "body1": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "14px",
          "fontWeight": "400",
          "letterSpacing": "0.02em",
          "lineHeight": "20px",
        },
        "body1Next": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "1rem",
          "fontWeight": 400,
          "lineHeight": 1.5,
        },
        "body2": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "12px",
          "fontWeight": "400",
          "letterSpacing": "0.02em",
          "lineHeight": "16px",
        },
        "body2Next": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "0.875rem",
          "fontWeight": 400,
          "lineHeight": 1.5,
        },
        "button": Object {
          "color": "#FFFFFF",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "14px",
          "fontWeight": "600",
          "letterSpacing": "0.02em",
          "lineHeight": "21px",
          "textTransform": "uppercase",
        },
        "buttonNext": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "lineHeight": 1.75,
          "textTransform": "uppercase",
        },
        "caption": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "0.75rem",
          "fontWeight": 400,
          "lineHeight": 1.66,
        },
        "captionNext": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "0.75rem",
          "fontWeight": 400,
          "lineHeight": 1.66,
        },
        "disabled": Object {
          "color": "#999999",
          "fontSize": "12px",
          "fontWeight": "400",
          "letterSpacing": "0.02em",
          "lineHeight": "16px",
        },
        "display1": Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "2.125rem",
          "fontWeight": 400,
          "lineHeight": "1.20588em",
        },
        "display2": Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "2.8125rem",
          "fontWeight": 400,
          "lineHeight": "1.13333em",
          "marginLeft": "-.02em",
        },
        "display3": Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "3.5rem",
          "fontWeight": 400,
          "letterSpacing": "-.02em",
          "lineHeight": "1.30357em",
          "marginLeft": "-.02em",
        },
        "display4": Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "7rem",
          "fontWeight": 300,
          "letterSpacing": "-.04em",
          "lineHeight": "1.14286em",
          "marginLeft": "-.04em",
        },
        "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
        "fontSize": 14,
        "fontWeightLight": 300,
        "fontWeightMedium": 500,
        "fontWeightRegular": 400,
        "h1": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "32px",
          "fontWeight": "600",
          "letterSpacing": "0.02em",
          "lineHeight": "40px",
        },
        "h2": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "32px",
          "fontWeight": "400",
          "letterSpacing": "0.02em",
          "lineHeight": "40px",
        },
        "h3": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "22px",
          "fontWeight": "600",
          "letterSpacing": "0.02em",
          "lineHeight": "30px",
        },
        "h4": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "22px",
          "fontWeight": "400",
          "letterSpacing": "0.02em",
          "lineHeight": "30px",
        },
        "h5": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "1.5rem",
          "fontWeight": 400,
          "lineHeight": 1.33,
        },
        "h6": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "1.25rem",
          "fontWeight": 500,
          "lineHeight": 1.6,
        },
        "headline": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "1.5rem",
          "fontWeight": 400,
          "lineHeight": "1.35417em",
        },
        "info": Object {
          "color": "#414141",
          "fontSize": "12px",
          "fontWeight": "400",
          "letterSpacing": "0.02em",
          "lineHeight": "16px",
        },
        "overline": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "0.75rem",
          "fontWeight": 400,
          "lineHeight": 2.66,
          "textTransform": "uppercase",
        },
        "pxToRem": [Function],
        "round": [Function],
        "sliderTrack": Object {
          "color": "#414141",
          "fontSize": "10px",
          "fontWeight": "400",
          "letterSpacing": "0.02em",
          "lineHeight": "12px",
        },
        "subheading": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "1rem",
          "fontWeight": 400,
          "lineHeight": "1.5em",
        },
        "subtitle1": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "14px",
          "fontWeight": "600",
          "letterSpacing": "0.02em",
          "lineHeight": "20px",
        },
        "subtitle2": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "12px",
          "fontWeight": "600",
          "letterSpacing": "0.02em",
          "lineHeight": "16px",
        },
        "suppressDeprecationWarnings": true,
        "title": Object {
          "color": "#414141",
          "fontFamily": "'Open Sans', 'Arial', 'Helvetica', sans-serif",
          "fontSize": "1.3125rem",
          "fontWeight": 500,
          "lineHeight": "1.16667em",
        },
        "useNextVariants": true,
        "visualization": Object {
          "color": "#414141",
          "fontSize": "10px",
          "fontWeight": "400",
          "letterSpacing": "0.02em",
          "lineHeight": "15px",
        },
      },
      "zIndex": Object {
        "appBar": 1100,
        "drawer": 1200,
        "mobileStepper": 1000,
        "modal": 1300,
        "snackbar": 1400,
        "tooltip": 1500,
      },
    }
  }
>
  <WithStyles(CssBaseline) />
  <ContextProvider
    value={
      Object {
        "router": null,
      }
    }
  >
    <WithStyles(Main)>
      Modal Content
    </WithStyles(Main)>
  </ContextProvider>
</MuiThemeProviderOld>
`;
